pyam/core.py

```diff
@@ -1506,3 +1506,19 @@ def compare(left, right, left_label='left', right_label='right',
     if drop_close:
         ret = ret[~np.isclose(ret[left_label], ret[right_label], **kwargs)]
     return ret[[right_label, left_label]]
+
+
+def concat(dfs):
+    """Concatenate a series of dataframes together"""
+    if not hasattr(dfs, '__iter__'):
+        raise TypeError('Input data must be iterable')
+
+    df = None
+    for _df in dfs:
+        if not isinstance(_df, IamDataFrame):
```

Comment:
the function `append()` casts to `IamDataFrame`, so this part can be removed - with the added benefit of being able to concat from a list with `pd.DataFrames` or files.